src/mscorlib/src/System/String.cs

```diff
@@ -1695,29 +1695,104 @@ private String CtorCharCount(char c, int count)
         private static unsafe int wcslen(char *ptr)
         {
             char *end = ptr;
+            
+            // TODO: Test to see if switching to something like
+            // the 64-bit version would be beneficial on x86 as well
+
+            // First make sure our pointer is aligned on a word boundary
+            int alignment = IntPtr.Size - 1;
+
+            // If ptr is at an odd address, this loop will simply iterate all the way
```

Comment:
> It's also true that this would never hit an aligned address if it started at an odd address

That's what I was referring to; perhaps I should clarify the comment.